src/desktop/components/main_layout/header/view.coffee

```diff
@@ -134,12 +136,16 @@ module.exports = class HeaderView extends Backbone.View
   checkForPostSignupAction: ->
     postSignupAction = Cookies.get 'postSignupAction'
     if postSignupAction
-      Cookies.expire 'postSignupAction'
       return unless @currentUser
-      { action, objectId } = JSON.parse(postSignupAction)
+      { action, objectId, kind } = JSON.parse(postSignupAction)
       if action is 'save'
         @currentUser.initializeDefaultArtworkCollection()
         @currentUser.defaultArtworkCollection().saveArtwork objectId
+      else if action is 'follow' and kind?
+        following = new Following [], kind: kind
+        following.follow objectId
+
+      Cookies.expire 'postSignupAction'
```

Comment:
Since following/saving is idempotent, I moved the cookie expiry to be the very last thing.